ui/src/contexts/Application.jsx

```diff
@@ -173,112 +163,70 @@ export default function Provider({ children }) {
-          console.error('got watchBrands err', err);
+  const setWalletP = async bridge => {
+    walletP = bridge;
+
+    await refreshConfigFromWallet(walletP);
+    const {
+      INSTALLATION_BOARD_ID,
+      INSTANCE_BOARD_ID,
+      RUN_ISSUER_BOARD_ID,
+      AMM_INSTALLATION_BOARD_ID,
+      AMM_INSTANCE_BOARD_ID,
+      AMM_NAME,
+    } = dappConfig;
+
+    const zoe = E(walletP).getZoe();
+    const board = E(walletP).getBoard();
+
+    await Promise.all([
+      setupTreasury(dispatch, brandToInfo, zoe, board, INSTANCE_BOARD_ID),
+      setupAMM(dispatch, brandToInfo, zoe, board, AMM_INSTANCE_BOARD_ID),
+    ]);
+
+    // The moral equivalent of walletGetPurses()
+    async function watchPurses() {
+      const pn = E(walletP).getPursesNotifier();
+      for await (const purses of iterateNotifier(pn)) {
+        dispatch(setPurses(purses));
+      }
+    }
+    watchPurses().catch(err =>
+      console.error('FIGME: got watchPurses err', err),
+    );
+
+    async function watchBrands() {
+      console.log('BRANDS REQUESTED');
+      const issuersN = E(walletP).getIssuersNotifier();
+      for await (const issuers of iterateNotifier(issuersN)) {
+        updateBrandPetnames({
+          dispatch,
+          brandToInfo,
+          issuersFromNotifier: issuers,
         });
-        await Promise.all([
-          E(walletP).suggestInstallation('Installation', INSTALLATION_BOARD_ID),
-          E(walletP).suggestInstance('Instance', INSTANCE_BOARD_ID),
-          E(walletP).suggestInstallation(
-            `${AMM_NAME}Installation`,
-            AMM_INSTALLATION_BOARD_ID,
-          ),
-          E(walletP).suggestInstance(
-            `${AMM_NAME}Instance`,
-            AMM_INSTANCE_BOARD_ID,
-          ),
-          E(walletP).suggestIssuer('RUN', RUN_ISSUER_BOARD_ID),
-        ]);
-
-        watchOffers(dispatch, INSTANCE_BOARD_ID);
-      },
-      onDisconnect() {
-        dispatch(setConnected(false));
-        walletAbort && walletAbort();
-        dispatch(resetState());
-      },
-      onMessage(data) {
-        const obj = JSON.parse(data);
-        walletDispatch && walletDispatch(obj);
-      },
+      }
+    }
+    watchBrands().catch(err => {
+      console.error('got watchBrands err', err);
     });
-    return deactivateWebSocket;
-  }, []);
+    await Promise.all([
+      E(walletP).suggestInstallation('Installation', INSTALLATION_BOARD_ID),
+      E(walletP).suggestInstance('Instance', INSTANCE_BOARD_ID),
+      E(walletP).suggestInstallation(
+        `${AMM_NAME}Installation`,
+        AMM_INSTALLATION_BOARD_ID,
+      ),
+      E(walletP).suggestInstance(`${AMM_NAME}Instance`, AMM_INSTANCE_BOARD_ID),
+      E(walletP).suggestIssuer('RUN', RUN_ISSUER_BOARD_ID),
+    ]);
+
+    watchOffers(dispatch, INSTANCE_BOARD_ID);
+  };
 
   return (
     <ApplicationContext.Provider value={{ state, dispatch, walletP }}>
       {children}
+      <WalletConnection setWalletP={setWalletP} dispatch={dispatch} />
```

Comment:
The wallet connection can't get it's props from the context or else there'd be a circular dependency